bash-specific syntax in build.sh (maybe others) with shebang line `#!/bin/sh`

So if the build image is switched to something with sh symlinked to others (eg. dash, in google-go.pkg.dev/golang), build fails.

```
make all 
# building for linux/amd64
./build/build.sh: 19: set: Illegal option -o pipefail
make: *** [Makefile:157: go-build] Error 2
```

We should consider removing bash-specific syntax, or declaring the shebang line as `#!/bin/bash`.

--- a/build/build.sh
+++ b/build/build.sh
@@ -16,7 +16,6 @@
 
 set -o errexit
 set -o nounset
-set -o pipefail
 
 if [ -z "${OS:-}" ]; then
     echo "OS must be set"
@@ -36,7 +35,7 @@ export GOARCH="${ARCH}"
 export GOOS="${OS}"
 export GO111MODULE=on
 
-if [[ "${DEBUG:-}" == 1 ]]; then
+if [ "${DEBUG:-}" = 1 ]; then
     # Debugging - disable optimizations and inlining
     gogcflags="all=-N -l"
     goasmflags=""

--- a/build/lint.sh
+++ b/build/lint.sh
@@ -16,7 +16,6 @@
 
 set -o errexit
 set -o nounset
-set -o pipefail
 
 export CGO_ENABLED=0
 export GO111MODULE=on
@@ -25,7 +24,7 @@ cd tools >/dev/null
 go install github.com/golangci/golangci-lint/cmd/golangci-lint
 cd - >/dev/null
 
-echo -n "Running golangci-lint: "
+printf "Running golangci-lint: "
 ERRS=$(golangci-lint run "$@" 2>&1 || true)
 if [ -n "${ERRS}" ]; then
     echo "FAIL"

--- a/build/test.sh
+++ b/build/test.sh
@@ -16,7 +16,6 @@
 
 set -o errexit
 set -o nounset
-set -o pipefail
 
 export CGO_ENABLED=0
 export GO111MODULE=on

